ui/components/app/signature-request-original/signature-request-original.component.js

```diff
@@ -105,12 +106,13 @@ export default class SignatureRequestOriginal extends Component {
   };
 
   renderBalance = () => {
-    const { conversionRate } = this.props;
     const {
+      conversionRate,
+      nativeCurrency,
       fromAccount: { balance },
-    } = this.state;
+    } = this.props;
```

Comment:
I think we should still get `fromAccount` from `this.state`

The component state is set on construction, but then not updated. And before this PR, the `fromAccount` was always just retrieved from state and not props. We should investigate why that was done that way and whether it requires correction, but that can be part of a separate PR.

Reply:
Actually I don't know, but I was checking other files [here](https://github.com/MetaMask/metamask-extension/blob/6187ab9b01ea757366a42d22b7292413c0f2b119/ui/pages/confirm-encryption-public-key/confirm-encryption-public-key.component.js#L112) also `props` is used instead of `state`.

Reply:
We can get fromAccount from `state`, but can't `nativeCurrency`.